tests/wrong-ref-not-string.yaml

```diff
@@ -0,0 +1,10 @@
+asyncapi: '2.2.0'
+info:
+  title: Account Service
+  version: 1.0.0
+  description: This service is in charge of processing user signups
+channels:
+  user/signedup:
+    subscribe:
+      message:
+        $ref: 5
```

Comment:
tbh, the tests that we have there, these are integration tests, and should not be used for testing type checking in just one of the functions imho. This is what unit tests are for, and there should be a unit test that checks `isExternalReference` function only. So both, string vs integer and my #/  would be better if tested as unit test. 

I'm completely fine if we put my comment aside and have separate issue to discuss proper approach for testing of the bundler

Reply:
In this case the issue is in the scope of the fix. The fix is narrowed **only** to adding type guard, so it is checking **only** type guard.
I will add checking the presence of `#/` in next PR, but it will be `chore:`.
(it is already checked indirectly by 'just working' in `./example/*` code, but I will add explicit check)
I was aligning with logic of previously written tests, but it might be time to rethink the theory of Bundler's testing indeed.

Reply:
Unit testing in common and of function `isExternalReference()` particularly was added by PR https://github.com/asyncapi/bundler/pull/99